backend/halloffame/tests.py

```diff
@@ -1,59 +1,106 @@
+                         {'user': self.user1.id, 'race': self.race1.id, 'guild': self.guild1.id})
 
         self.assertEqual(Hero.objects.count(), 1)
 
     def test_user_can_not_create_second_hero(self):
-        factories.HeroFactory(user=self.user1)
-        response = self.client.post(self.hero_list_url,
-                                    {'user': self.user1.id, 'race': self.race1.id, 'guild': self.guild1.id})
+        HeroFactory(user=self.user1)
+        self.client.post(self.hero_list_url,
+                         {'user': self.user1.id, 'race': self.race1.id, 'guild': self.guild1.id})
 
         self.assertEqual(Hero.objects.count(), 1)
 
     def test_create_new_hero_assigned_to_someone_else(self):
-        response = self.client.post(self.hero_list_url,
-                                    {'user': self.user2.id, 'race': self.race1.id, 'guild': self.guild1.id})
+        self.client.post(self.hero_list_url,
+                         {'user': self.user2.id, 'race': self.race1.id, 'guild': self.guild1.id})
 
         self.assertEqual(Hero.objects.count(), 0)
 
     def test_delete_someone_elses_hero(self):
-        factories.HeroFactory(user=self.user2)
-        response = self.client.delete(self.user2_url)
+        HeroFactory(user=self.user2)
+        self.client.delete(self.user2_url)
 
         self.assertEqual(Hero.objects.count(), 1)
 
     def test_user_can_change_guild(self):
-        factories.HeroFactory(user=self.user1)
+        HeroFactory(user=self.user1)
         response = self.client.patch(self.user1_url, {'user': self.user1.id, 'guild': self.guild2.id})
 
         self.assertEqual(response.status_code, status.HTTP_200_OK)
 
     def test_user_cant_change_someone_elses_guild(self):
-        factories.HeroFactory(user=self.user2)
+        HeroFactory(user=self.user2)
         response = self.client.patch(self.user2_url, {'user': self.user2.id, 'guild': self.guild2.id})
 
         self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+
+
+class HallOfFameFilteringTestCase(test.APITestCase):
+    @classmethod
+    def setUpTestData(cls):
+        cls.hero_list_url = reverse('hero-list')
+        cls.hero1 = HeroFactory()
+        cls.hero2 = HeroFactory()
+        cls.hero3 = HeroFactory()
+
+        cls.battle1 = Battle.objects.create(date=timezone.now())
+        cls.battle1.attendees.set([cls.hero1, cls.hero2])
+        cls.battle1.looser = cls.hero1
+        cls.battle1.save()
+
+        cls.battle2 = Battle.objects.create(date=timezone.now())
+        cls.battle2.attendees.set([cls.hero1, cls.hero3])
+        cls.battle2.looser = cls.hero1
+        cls.battle2.save()
+
+        cls.battle3 = Battle.objects.create(date=timezone.now(), is_looser_dead=True)
+        cls.battle3.attendees.set([cls.hero2, cls.hero3])
+        cls.battle3.looser = cls.hero3
+        cls.battle3.save()
+
+    def test_alive_heroes_filtering(self):
+        response = self.client.get(self.hero_list_url + '?is_alive=True')
```

Comment:
You could also do it this way:
```suggestion
        alive_heroes_url = reverse('hero-list', kwargs={'is_alive': 'True'}
        response = self.client.get(alive_heroes_url)
```

Reply:
?

Reply:
Ooops, actually I made a mistake, the extra arguments go in the `get` method and not `reverse`, here's the way to do it:
```
alive_heroes_url = reverse('hero-list')
response = self.client.get(alive_heroes_url, {'is_alive': 'True'})
```
Let me know if it works, and sorry for the confusion, my bad! :smile: